x/approvals/handler.go

```diff
@@ -0,0 +1,96 @@
+package approvals
+
+import (
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/errors"
+	"github.com/iov-one/weave/orm"
+	"github.com/iov-one/weave/x"
+	"github.com/iov-one/weave/x/nft"
+)
+
+const Admin = "admin"
+
+type AddApprovalMsgHandler struct {
+	auth   x.Authenticator
+	bucket orm.Bucket
+}
+
+func NewAddApprovalMsgHandler(auth x.Authenticator, bucket orm.Bucket) AddApprovalMsgHandler {
+	return AddApprovalMsgHandler{auth, bucket}
+}
+
+var _ weave.Handler = AddApprovalMsgHandler{}
+
+func (h AddApprovalMsgHandler) Check(ctx weave.Context, store weave.KVStore, tx weave.Tx) (weave.CheckResult, error) {
+	var res weave.CheckResult
+	_, _, err := h.validate(ctx, store, tx)
+	if err != nil {
+		return res, err
+	}
+
+	return res, nil
+}
+
+func (h AddApprovalMsgHandler) Deliver(ctx weave.Context, store weave.KVStore, tx weave.Tx) (weave.DeliverResult, error) {
+	var res weave.DeliverResult
+	msg, appr, err := h.validate(ctx, store, tx)
+	if err != nil {
+		return res, err
+	}
+
+	appr.UpdateApprovals(append(appr.GetApprovals(), msg.GetApproval()))
```

Comment:
This is also a bit flaky. They spent some time in the original version to make sure this was consistent. eg. no duplicates, reliable order. They should be sorted so one set of approvals will always have the same representation.